From fe29c871a32a3a865e8d8fd4f8466a63c24caf88 Mon Sep 17 00:00:00 2001
From: Roi Dayan <roid@mellanox.com>
Date: Wed, 28 Aug 2013 19:18:13 +0300
Subject: [PATCH] Fix possible segfault on logicalunit update

When updating logicalunit with tgtdadm without any params
tgtd will segfault when calling strncmp on NULL.
The call to strsep later is ok and will return NULL.

To reproduce create a target with a lun and run:
tgtadm -m logicalunit -o update --tid 1 --lun 1

Signed-off-by: Roi Dayan <roid@mellanox.com>
Signed-off-by: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
---
 usr/spc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/spc.c b/usr/spc.c
index 7ff7834..1d02db3 100644
--- a/usr/spc.c
+++ b/usr/spc.c
@@ -1710,7 +1710,7 @@ int lu_config(struct scsi_lu *lu, char *params, match_fn_t *fn)
 	attrs = &lu->attrs;
 	lu_vpd = attrs->lu_vpd;
 
-	if (!strncmp("targetOps", params, 9))
+	if (params && !strncmp("targetOps", params, 9))
 		params = params + 10;
 
 	while ((p = strsep(&params, ",")) != NULL) {
